fdbserver/BlobWorker.actor.cpp

```diff
@@ -5099,10 +5108,12 @@ ACTOR Future<Void> simForceFullMemory(Reference<BlobWorkerData> bwData) {
 ACTOR Future<Void> blobWorker(BlobWorkerInterface bwInterf,
                               ReplyPromise<InitializeBlobWorkerReply> recruitReply,
                               Reference<AsyncVar<ServerDBInfo> const> dbInfo) {
-	state Reference<BlobWorkerData> self(new BlobWorkerData(
-	    bwInterf.id(), dbInfo, openDBOnServer(dbInfo, TaskPriority::DefaultEndpoint, LockAware::True)));
+	state Database cx = openDBOnServer(dbInfo, TaskPriority::DefaultEndpoint, LockAware::True);
+	state Reference<BlobWorkerData> self(new BlobWorkerData(bwInterf.id(), dbInfo, cx));
 	self->id = bwInterf.id();
 	self->locality = bwInterf.locality;
+	DatabaseConfiguration config = wait(getDatabaseConfiguration(cx));
```

Comment:
Shall we pass encryption mode through recruitment request just like what CP and SS recruitment does?

Reply:
The way blobworker gets initalized is through the blob manager. This means we would need to pass the the encryption state to both the blob manager and blob worker (which requires more code changes that necessary since blob manager doesn't care about the encryption state). Since we are already fetching the database configuration here i feel like this is a simpler solution

Reply:
Sounds good. Can you add inline comment to describe why we are doing differently here?